packages/navigator/src/browser/open-editors-widget/navigator-open-editors-decorator-service.ts

```diff
@@ -0,0 +1,80 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, named } from '@theia/core/shared/inversify';
+import { TreeDecorator, AbstractTreeDecoratorService, TreeDecoration } from '@theia/core/lib/browser/tree/tree-decorator';
+import { Tree } from '@theia/core/lib/browser';
+import { ContributionProvider } from '@theia/core/lib/common';
+
+export const OpenEditorsTreeDecorator = Symbol('OpenEditorsTreeDecorator');
+
+@injectable()
+export class OpenEditorsTreeDecoratorService extends AbstractTreeDecoratorService {
```

Comment:
I'm a bit confused as to this file. 
1. why does it live in `navigator`, `navigator` should not know about `problems`.
2. I believe we want a `open-editor` decorator service which `problems` contributes to, like the following in `problems/`.
```ts
bind(NavigatorTreeDecorator).toService(ProblemDecorator);
```

Reply:
I've moved the suffix decorations that add the `workspace * path/to/file` to the problem decorator and added a conditional check if the decorated node belongs to the OpenEditorsWidget. This way the suffix decorations are only applied to open-editor nodes and not the navigator nodes. Moving that logic allows this file to be much less complex :)

Thanks for the suggestions and help with this issue!

The only downside is that I cannot find a way to aggregate the italicization provided by the `editor-preview-tree-decorator` into the suffix decoration, which causes a preview widget node to not have italicization on its suffix. I don't think this is really a problem and the cleaner code outweighs the minor UI flaw. This could also be addressed in a separate PR

![image](https://user-images.githubusercontent.com/62660714/126011162-cffafe00-80dd-4e64-ab03-b18c71fc7209.png)